File: 1-two-sum/two-sum.php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $target
     * @return Integer[]
     */
    function twoSum($nums, $target) {
        $n=count($nums);
        $map = [];
        for($i = 0 ; $i < count($nums) ; $i++){
            if(isset($map[$target - $nums[$i]])){
                return array($map[$target - $nums[$i]], $i);
            }
            $map[$nums[$i]] = $i;
        }
        return array();
    }
}

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int,int> mp;
        for(int i = 0 ; i < nums.size() ; i++){
            if(mp.contains(target - nums[i])){
                return vector<int>{mp[target - nums[i]],i};
            }
            mp[nums[i]] = i;
        }
        return vector<int>();
    }
};